services/src/main/java/org/keycloak/services/clientregistration/ClientRegistrationAuth.java

```diff
@@ -283,13 +293,36 @@ public ClientInitialAccessModel getInitialAccessModel() {
 
     private boolean hasRole(String... roles) {
         try {
-            if (jwt.getIssuedFor().equals(Constants.ADMIN_CLI_CLIENT_ID)
-                    || jwt.getIssuedFor().equals(Constants.ADMIN_CONSOLE_CLIENT_ID)) {
-                return hasRoleInModel(roles);
 
-            } else {
-                return hasRoleInToken(roles);
+            //support for lightweight access token
+            if (jwt.getSubject() == null) {
+                String sid = (String) jwt.getOtherClaims().get("sid");
+                if (sid != null) {
+                    final String issuedFor = jwt.getIssuedFor();
+                    UserSessionProvider sessions = session.sessions();
+                    UserSessionModel userSession = sessions.getUserSession(realm, sid);
+                    if (userSession == null) {
+                        userSession = sessions.getOfflineUserSession(realm, sid);
+                    }
+
+                    //get client session
+                    ClientModel client = realm.getClientByClientId(issuedFor);
```

Comment:
I guess it can still happen that `userSession` is null after this? Same for `client` and `clientSession` variables below? Is it possible to handle his better way than NullPointerException? Maybe it is OK to just add debug log message and not set anything into `jwt` for this case.